### PR TITLE
[Storage] support FederatedIdentityClientId

### PR DESCRIPTION
**Pull request by blueww**
## Description

Will confirm with feature team for the release schedule, them remove "Do Not merge" tag.

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
    - the test environment is very complex, need 2 Azure AD (with login credential). So will cover test manually.
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1114
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by blueww** [comment]
Not for now. Feature team would like it be in on April

File: src/Storage/Storage.Management/help/New-AzStorageAccount.md [hunk 1]
```diff
@@ -313,9 +313,33 @@ AAD
...
+KeyVaultUri                   : https://mykeyvault.vault.azure.net:443
+CurrentVersionedKeyIdentifier : https://mykeyvault.vault.azure.net/keys/wrappingKey/8e74036e0d534e58b3bd84b319e31d8f
+LastKeyRotationTimestamp      : 3/3/2022 2:07:34 AM
 ```
-New-AzStorageAccount -ResourceGroupName "MyResourceGroup" -AccountName "mystorageaccount" -SkuName Standard_LRS  -Location centraluseuap -Kind StorageV2 -DnsEndpointType AzureDnsZone 
+
+This command creates a storage account with Keyvault from another tenant (access Keyvault with FederatedClientId).
+
+### Example 18: Create account with DnsEndpointType as AzureDnsZone
+```
```
Comment (isra-fel) [inline src/Storage/Storage.Management/help/New-AzStorageAccount.md:341]:
```suggestion
```powershell
```

**Review by isra-fel — COMMENTED** [review]
And please resolve the conflicts.

**Review by isra-fel — APPROVED** [review]
LGTM!

**Pull request merged**